tests/epyccel/test_epyccel_lists.py

```diff
@@ -0,0 +1,84 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring
+import pytest
+from  pyccel.epyccel import epyccel
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason="Function in function not implemented in fortran"),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = [
+            pytest.mark.xfail(reason="Function in function not implemented in C"),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+
+def test_pop_last_element(language) :
+    def po():
+        a = [1,3,45]
+        return a.pop()
+    re = epyccel(po, language = language)
+    assert isinstance(po(), type(re()))
+    assert po() == re()
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason="method pop for list not implemented in fortran"),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = [
+            pytest.mark.xfail(reason="method pop for list not implemented in C"),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+
+def test_pop_specific_index(language) :
+    def po():
+        a = [1j,3j,45j]
+        return a.pop(1)
+    re = epyccel(po, language = language)
+    assert isinstance(po(), type(re()))
+    assert po() == re()
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason="method pop for list not implemented in fortran"),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = [
+            pytest.mark.xfail(reason="method pop for list not implemented in C"),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+
+def test_pop_negative_index(language) :
+    def po():
+        a = [1j,3j,45j]
+        return a.pop(-1)
+    re = epyccel(po, language = language)
+    assert isinstance(po(), type(re()))
+    assert po() == re()
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason="method pop for list not implemented in fortran"),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = [
+            pytest.mark.xfail(reason="method pop for list not implemented in C"),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+
+def test_pop_2(language) :
+    def po():
+        a = [1.7,2.7,45.0]
+        a.pop()
+        return a.pop(-1)
+    re = epyccel(po, language = language)
+    assert isinstance(po(), type(re()))
+    assert po() == re()
```

Comment:
This file contains lots of repeat calls. This could be problematic if the function or its wrapper is costly. Why not save the results?

Additionally the names are not very explicit. What does `re` stand for?

```suggestion
def test_pop_2(language) :
    def pop_2():
        a = [1.7,2.7,45.0]
        a.pop()
        return a.pop(-1)
    pop_2_epyc = epyccel(po, language = language)
    python_result = pop_2()
    pyccel_result = pop_2_epyc()
    assert isinstance(python_result, type(pyccel_result))
    assert python_result == pyccel_result
```

Reply:
Fixed 8e921a7c9a042604974a2e1d53a18cfc8b3e8d3f